docs/how-tos/build-a-plugin.md

```diff
@@ -0,0 +1,192 @@
+
+```json
+{
+  "/ENTRY[my_entry]/INSTRUMENT[my_instrument]/SOURCE[my_source]/type": "None"
+}
+```
+
+#### Attributes
+For attributes defined in the NXDL, the reader template dictionary will have the assosciated key with a "@" prefix to the attributes name at the end of the path:
+
+```json
+{
+  "/entry/instrument/source/@attribute": "None"
+}
+```
+
+#### Units
+If there is a field defined in the NXDL, the converter expects a filled in /data/@units entry in the template dictionary corresponding to the right /data field unless it is specified as NX_UNITLESS in the NXDL. Otherwise, you will get an exception.
+
+```json
+{
+  "/ENTRY[my_entry]/INSTRUMENT[my_instrument]/SOURCE[my_source]/data": "None",
+  "/ENTRY[my_entry]/INSTRUMENT[my_instrument]/SOURCE[my_source]/data/@units": "Should be set to a string value"
+}
+```
+
+#### Links
+You can also define links by setting the value to sub dictionary object with key `link`:
+
+```python
+template["/entry/instrument/source"] = {"link": "/path/to/source/data"}
+```
+
+### Building off the BaseReader
+When building off the [`BaseReader`](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/readers/base/reader.py), the developer has the most flexibility. Any new reader must implement the `read` function, which must return a filled template object.
+
+
+### Building off the MultiFormatReader
+While building on the ```BaseReader``` allows for the most flexibility, in most cases it is desirable to implement a reader that can read in multiple file formats and then populate the template based on the read data. For this purpose, `pynxtools` has the [**`MultiFormatReader`**](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/readers/multi/reader.py), which can be readily extended for your own data.
+
+You can find an extensive how-to guide to build off the `MultiFormatReader` [here](./use-multi-format-reader.md).
+
+
+## Calling the reader from the command line
+
+The pynxtools converter allows extending support to other data formats by allowing extensions called readers.
+The converter provides a dev platform to build a NeXus compatible reader by providing checking against a chosen NeXus Application Definition.
+
+The dataconverter can be executed using:
+```console
+user@box:~$ dataconverter --reader mydatareader --nxdl NXmynxdl --output path_to_output.nxs
+```
+Here, the ``--reader`` flag must match the reader name defined in `[project.entry-points."pynxtools.reader"]` in the pyproject.toml file. The NXDL name passed to ``--nxdl``must be a valid NeXus NXDL/XML file in `pynxtools.definitions`.
+
+Aside from this default structure, there are many more flags that can be passed to the dataconverter call. Here is an ouput if its ```help``` call:
+```console
```

Comment:
single source principle, beyond this line because if the click options docstrings change they have to be changed also in this documentation here, instead one could use a python script whereby the current code is fished and then added here programmatically, avoiding a potential source of inconsistence between actual code and its documentation.

Reply:
I changed it to include a file that is dynamically created using a python script. Could be cleaner, but works for now.